Platon/backend/app/search_engine/views.py

```diff
@@ -308,7 +329,114 @@ def get(self):
+            for token, score in tokens:
+                try:
+                    query = '(LOWER(title) REGEXP ".*{0}.*" \
+                    OR LOWER(acronym) REGEXP ".*{0}.*" OR LOWER(location) REGEXP ".*{0}.*")'  \
+                    .format(token)
+                    sql_statement = "SELECT * FROM upcoming_events WHERE {}".format(query)
+                    result = db.engine.execute(sql_statement)
+                    for upcoming_event in result:
+                        id_list = [i[0] for i in result_id_score]
+                        if upcoming_event[0] not in id_list:
+                            if upcoming_event[-1] == 0:
+                                continue
+                            result_list.append({"id": upcoming_event[0], "title": upcoming_event[1], "acronym": upcoming_event[2],
+                                                "location": upcoming_event[3], "date": upcoming_event[4],
+                                                "deadline": upcoming_event[5], "link": upcoming_event[6]})
+                            result_id_score.append((upcoming_event[0], score))
+                        else:
+                            id_index = id_list.index(upcoming_event[0])
+                            result_id_score[id_index] = (upcoming_event[0], result_id_score[id_index][1] + score)
+                except:
+                    return make_response(jsonify({"error": "Database Connection Problem."}), 500)
 
+            # Search tokens for deadline match
+            deadline_filter_list = []
+            if form.deadline_filter.data != '':
+                for result in result_list:
+                    event_deadline = result.get("deadline")
+                    event_deadline = SearchEngine.remove_punctuation(event_deadline)
+                    remove_digits = str.maketrans('', '', digits)
+                    filter_data = form.deadline_filter.data.translate(remove_digits)
+                    filter_data = SearchEngine.remove_punctuation(filter_data)
+                    if filter_data in event_deadline:
+                        deadline_filter_list.append(result)
+                    result_list = deadline_filter_list
+            # Search tokens for date match
+            date_filter_list = []
+            if form.date_filter.data != '':
+                for result in result_list:
+                    event_date = result.get("date")
+                    event_date = SearchEngine.remove_punctuation(event_date)
+                    remove_digits = str.maketrans('', '', digits)
+                    filter_data = form.date_filter.data.translate(remove_digits)
+                    filter_data = SearchEngine.remove_punctuation(filter_data)
+                    if filter_data in event_date:
+                        date_filter_list.append(result)
+                    result_list = date_filter_list
+            sorted_result_list = []
```

Comment:
I really can not understand your filtering algorithm. And I the,nk this part does not apply meaningful filtering to dates. Please update the date filtering part by converting strings to datetime objects by using the approach on the following link:
https://www.educative.io/edpresso/how-to-convert-a-string-to-a-date-in-python